### ISSUE TITLE
CI: k8s-1.8 cluster not ready before running preflight checks

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#6441]
`CI:  k8s-1.8.K8sHealthTest checks cilium-health status between nodes`

In PR #6435 I hit an issue where in the k8s-1.8 VM, the preflight checks failed in the first test run. Inspecting the logs, Cilium has restarted multiple times. The most recent restart contained these final log messages:

```
2018-12-11T17:48:42.665833416Z level=fatal msg="Unable to connect to kvstore" error="rpc error: code = Unknown desc = " subsys=kvstore
2018-12-11T17:48:42.668942723Z level=fatal msg="Agent pipe unexpectedly closed, shutting down" subsys=cilium-node-monitor
```

It seems like perhaps the initial bootstrapping in this VM was not complete before starting to run the {preflight checks, tests}. Subsequent tests seem to work fine.

Failure:
```
Stacktrace
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:340
cilium pre-flight checks failed
Expected
    <*errors.errorString | 0xc4203c0940>: {
        s: "CiliumPreFlightCheck error: Timeout reached: PreflightCheck failed: Cluster connectivity is unhealthy on 'cilium-76f8g': Exitcode: 255 \nStdout:\n \t \nStderr:\n \t Error: Cannot get status/probe: context deadline exceeded\n\t \n\t command terminated with exit code 255\n\t \n",
    }
to be nil
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/k8sT/assertionHelpers.go:110
Standard Output
Number of 'context deadline exceeded' in logs: 0
Number of 'level=error' in logs: 0
⚠️  Number of 'level=warning' in logs: 7
Number of 'Cilium API handler panicked' in logs: 0
⚠️  Number of 'Goroutine took lock for more than' in logs: 40
Cilium pods: [cilium-76f8g cilium-t9bxn]
Netpols loaded: 
CiliumNetworkPolicies loaded: 
Endpoint Policy Enforcement:
     cilium-etcd-nqhztbf2vc             =>   none
     cilium-etcd-rnp47fkx4t             =>   none
     etcd-operator-696fb6d99-4kwnd      =>   none
     kube-dns-6f4c8dbcd4-7q7jd          =>   none
     prometheus-core-5c9fbf5cc6-27pj6   =>   none
     cilium-etcd-4bpqs7gwhb             =>   none
     cilium-etcd-5rvdrx2mvb             =>   none
     cilium-etcd-hj4k7x76kr             =>   none
Cilium agent 'cilium-76f8g': Status: Ok  Health: Ok Nodes "k8s2 k8s1" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 31 Failed 0
Cilium agent 'cilium-t9bxn': Status: Ok  Health: Ok Nodes "k8s2 k8s1" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 27 Failed 0

Standard Error
STEP: Installing DNS Deployment
STEP: Deploying etcd-operator
STEP: Installing Cilium
STEP: Performing Cilium preflight check
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that cilium-health status is healthy
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that cilium-health status is healthy
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that cilium-health status is healthy
===================== TEST FAILED =====================
cmd: kubectl get pods -o wide --all-namespaces
Exitcode: 0 
Stdout:
 	 NAMESPACE     NAME                                   READY     STATUS    RESTARTS   AGE       IP              NODE
	 kube-system   cilium-76f8g                           1/1       Running   4          6m        192.168.36.12   k8s2
	 kube-system   cilium-etcd-5rvdrx2mvb                 1/1       Running   0          28s       10.10.0.148     k8s1
	 kube-system   cilium-etcd-nqhztbf2vc                 1/1       Running   0          11s       10.10.1.91      k8s2
	 kube-system   cilium-etcd-operator-8796fb686-gj2zs   1/1       Running   0          6m        192.168.36.12   k8s2
	 kube-system   cilium-etcd-rnp47fkx4t                 1/1       Running   0          36s       10.10.1.2       k8s2
	 kube-system   cilium-t9bxn                           1/1       Running   4          6m        192.168.36.11   k8s1
	 kube-system   etcd-k8s1                              1/1       Running   0          10m       192.168.36.11   k8s1
	 kube-system   etcd-operator-696fb6d99-4kwnd          1/1       Running   0          5m        10.10.1.184     k8s2
	 kube-system   kube-apiserver-k8s1                    1/1       Running   0          9m        192.168.36.11   k8s1
	 kube-system   kube-controller-manager-k8s1           1/1       Running   0          10m       192.168.36.11   k8s1
	 kube-system   kube-dns-6f4c8dbcd4-7q7jd              3/3       Running   3          9m        10.10.0.47      k8s1
	 kube-system   kube-proxy-k29zq                       1/1       Running   0          6m        192.168.36.12   k8s2
	 kube-system   kube-proxy-klqps                       1/1       Running   0          9m        192.168.36.11   k8s1
	 kube-system   kube-scheduler-k8s1                    1/1       Running   0          10m       192.168.36.11   k8s1
	 monitoring    prometheus-core-5c9fbf5cc6-27pj6       1/1       Running   0          6m        10.10.1.190     k8s2
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-76f8g -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                              IPv6                   IPv4          STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                                
	 14532      Disabled           Disabled          101        k8s:app=etcd                                             f00d::a0a:100:0:5649   10.10.1.91    ready   
	                                                            k8s:etcd_cluster=cilium-etcd                                                                          
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
	                                                            k8s:io.cilium/app=etcd-operator                                                                       
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
	 27025      Disabled           Disabled          26413      k8s:app=prometheus                                       f00d::a0a:100:0:8e2    10.10.1.190   ready   
	                                                            k8s:component=core                                                                                    
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=prometheus-k8s                                                
	                                                            k8s:io.kubernetes.pod.namespace=monitoring                                                            
	 29437      Disabled           Disabled          100        k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0a:100:0:73a8   10.10.1.184   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=cilium-etcd-sa                                                
	                                                            k8s:io.cilium/app=etcd-operator                                                                       
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
	 49303      Disabled           Disabled          4          reserved:health                                          f00d::a0a:100:0:f5c6   10.10.1.42    ready   
	 63735      Disabled           Disabled          101        k8s:app=etcd                                             f00d::a0a:100:0:be14   10.10.1.2     ready   
	                                                            k8s:etcd_cluster=cilium-etcd                                                                          
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
	                                                            k8s:io.cilium/app=etcd-operator                                                                       
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
	 
Stderr:
 	 level=info msg="Detected MTU 1500" subsys=mtu
	 

cmd: kubectl exec -n kube-system cilium-t9bxn -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                        IPv6                 IPv4          STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                        
	 29704      Disabled           Disabled          102        k8s:io.cilium.k8s.policy.cluster=default           f00d::a0a:0:0:26c    10.10.0.47    ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=kube-dns                                              
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                   
	                                                            k8s:k8s-app=kube-dns                                                                          
	 44695      Disabled           Disabled          101        k8s:app=etcd                                       f00d::a0a:0:0:28b6   10.10.0.148   ready   
	                                                            k8s:etcd_cluster=cilium-etcd                                                                  
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                      
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                               
	                                                            k8s:io.cilium/app=etcd-operator                                                               
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                   
	 61072      Disabled           Disabled          4          reserved:health                                    f00d::a0a:0:0:d6cb   10.10.0.21    ready   
	 
Stderr:
 	 level=info msg="Detected MTU 1500" subsys=mtu
	 

===================== Existing AfterFailed =====================

[[ATTACHMENT|9985ac59_K8sHealthTest_checks_cilium-health_status_between_nodes.zip]]
```

Link: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/8540/execution/node/96/log/?consoleFull
Zip: 
[9985ac59_K8sHealthTest_checks_cilium-health_status_between_nodes (1).zip](https://github.com/cilium/cilium/files/2668913/9985ac59_K8sHealthTest_checks_cilium-health_status_between_nodes.1.zip)

**Comment by joestringer** [comment]
Likely duplicate of #6370.
